src/vsphere_cpi/lib/cloud/vsphere/attach_tag_to_vm.rb

```diff
@@ -0,0 +1,243 @@
+require 'cloud/vsphere/logger'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module  VSphereCloud
+  module TaggingTag
+
+    class VCenterBadEntityError < StandardError
```

Comment:
What's the use of this if you have `VCenterNoTagFoundError` and `VCenterNoCategoryExitsError `

Reply:
'VCenterNoCategoryExitsError' avoid to traverse the categories in tag_hash, when there is no categories on vCenter, we just finish attach tag steps.

Similar with  'VCenterNoTagFoundError', when try to attach one or more tags in a category, we first use the category id got a list of tags under it, when the list is empty, throw out the message and skip check tags in category.

 'VCenterBadEntityError' refers to a single category or tag check result. 

These 3 throws different messages, that why we need 3 classes.